src/strategy/StrategyV1.sol

```diff
@@ -0,0 +1,678 @@
+        );
+
+        if (!isRewardReceived) {
+            ILiquidityGauge(gauge).claim_rewards(address(locker));
+        }
+
+        for (i = 0; i < 8;) {
+            extraRewardToken = extraRewardTokens[i];
+            if (extraRewardToken == address(0)) break;
+            uint256 claimed;
+            if (!isRewardReceived) {
+                claimed = ERC20(extraRewardToken).balanceOf(address(locker)) - snapshotLockerRewardBalances[i];
+                if (claimed != 0) {
+                    // Transfer the freshly rewards from the locker to this contract.
+                    _transferFromLocker(extraRewardToken, address(this), claimed);
+                }
+            }
+
+            if (extraRewardToken == rewardToken) {
+                _rewardTokenClaimed += claimed;
+            } else {
+                claimed = ERC20(extraRewardToken).balanceOf(address(this));
+                if (claimed != 0) {
+                    // Distribute the extra reward token.
+                    ILiquidityGauge(rewardDistributor).deposit_reward_token(extraRewardToken, claimed);
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- VIEW FUNCTIONS
+    //////////////////////////////////////////////////////
+
+    function balanceOf(address asset) public view virtual returns (uint256) {
+        // Get the gauge address
+        address gauge = gauges[asset];
+        if (gauge == address(0)) revert ADDRESS_NULL();
+
+        return ILiquidityGauge(gauge).balanceOf(address(locker));
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- INTERNAL HELPER FUNCTIONS
+    //////////////////////////////////////////////////////
+
+    function _transferFromLocker(address asset, address recipient, uint256 amount) internal virtual {
+        locker.safeExecute(asset, 0, abi.encodeWithSignature("transfer(address,uint256)", recipient, amount));
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- MIGRATION LOGIC
+    //////////////////////////////////////////////////////
+
+    /// @notice If we ever decide to migrate the strategy, we can use this function to migrate the LP token.
+    /// @dev Only callable by the vault.
+    /// @param asset Address of LP token to migrate.
+    /// @dev Built only to support the old implementation of the vault, but it will be killed.
+    function migrateLP(address asset) public virtual onlyVault {}
+
+    //////////////////////////////////////////////////////
+    /// --- FACTORY STRATEGY SETTERS
+    //////////////////////////////////////////////////////
+
+    /// @notice Toogle vault status
+    /// @param vault Address of the vault to toggle
+    function toggleVault(address vault) external onlyGovernanceOrFactory {
+        if (vault == address(0)) revert ADDRESS_NULL();
+        vaults[vault] = !vaults[vault];
+    }
```

Comment:
If there is no hard legacy already relying on this signature, I would argue that using explicit setters like `activeVault` and `deactiveVault` is less error-prone than relying on a toggle function. Definitely my recommendation here.